smac_planner/src/a_star.cpp

```diff
@@ -365,6 +384,84 @@ bool AStarAlgorithm<NodeT>::isGoal(NodePtr & node)
 }
 
 template<>
+AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(const NodePtr & node, const ValidityChecker & checker)
+{
+  if (_motion_model == MotionModel::DUBIN || _motion_model == MotionModel::REEDS_SHEPP)
+  {
+    std::vector<std::pair<NodePtr, Coordinates>> possible_nodes;
+    const NodePtr & goal = getGoal();
+    ompl::base::ScopedState<> from(node->motion_table.state_space), to(node->motion_table.state_space), s(node->motion_table.state_space);
+    const NodeSE2::Coordinates & node_coords = node->pose;
+    const NodeSE2::Coordinates & goal_coords = goal->pose;
+    from[0] = node_coords.x;
+    from[1] = node_coords.y;
+    from[2] = node_coords.theta * node->motion_table.bin_size;
+    to[0] = goal_coords.x;
+    to[1] = goal_coords.y;
+    to[2] = goal_coords.theta * node->motion_table.bin_size;
+
+    float d = node->motion_table.state_space->distance(from(), to());
+    NodePtr prev(node);
+    // A move of sqrt(2) is guaranteed to be in a new cell
+    constexpr float sqrt_2 = std::pow(2., 0.5);
```

Comment:
I know this is stupid, but `sqrt(2)` is much faster. `pow` uses a heavier iterative version. When you work with even powers or when it can be a normal root, `sqrt` or direct multiplication is much faster.

Reply:
Yes, I don't know why I didn't do that. Brain on autopilot, I think.
Fixed.